modules/storages/app/common/storages/peripherals/storage_interaction/storage_commands.rb

```diff
@@ -30,26 +30,13 @@ module Storages::Peripherals::StorageInteraction
   class StorageCommands
     using ::Storages::Peripherals::ServiceResultRefinements
 
-    def initialize(uri:, provider_type:, username:, password:)
-      @uri = uri
-      @provider_type = provider_type
-      @username = username
-      @password = password
+    def initialize(storage)
+      @storage = storage
     end
 
     def set_permissions_command
-      case @provider_type
-      when ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD
-        ServiceResult.success(
-          result: ::Storages::Peripherals::StorageInteraction::Nextcloud::SetPermissionsCommand.new(
-            base_uri: @uri,
-            username: @username,
-            password: @password
-          )
-        )
-      else
-        raise ArgumentError
-      end
+      command_class = "::Storages::Peripherals::StorageInteraction::#{@storage.short_provider_type.capitalize}::SetPermissionsCommand".constantize
```

Comment:
I would suggest to use it for storage queries as well. @apfohl wdyt?

Reply:
I see that it would make this part simpler. But keep in mind if you start to construct class names with strings like that, no IDE would be able to find this symbol ever again. To e.g. auto-refactor it to something else. It would need to be done manually every time and every where. You can leave it for now, but I would like to find a better solution, that is using Ruby syntax instead of strings.

Reply:
Yes, that is a fair point. My intention is to avoid touching this part of code with new provider_type arrival.
It seems hard to achieve without looking up for constants somehow during runtime or bootstrap of the app. So, I returned the old version with `case ... when ... end` statement.